bin/busted

```diff
@@ -59,6 +59,14 @@ require("kong.globalpatches")({
   rbusted = true
 })
 
+-- some libraries used in test like spec/helpers
+-- calls cosocket in module level, and as LuaJIT's
+-- `require` is implemented in C, this throws
+-- "attempt to yield across C-call boundary" error
+-- the following pure-lua implementation is to bypass
+-- this limitation, without need to modify all tests
+_G.require = require "spec.require".require
```

Comment:
we've seen these errors as well with Kong itself many times, shouldn't we add this to the global-patches instead? then both Kong and tests will be free of this nagging error.

Reply:
Moved to globalpatches

Reply:
Wouldn't it be better to use luarocks to manage this dependency, rather than pulling a separate copy into our codebase?

Reply:
I agree with @ms2008, we usually don't put 3rd party modules under `kong/` directory. Either move it to a subdirectory, or use luarocks.

On the other side, since we move it to globalpatches not just in tests, please add unit test for `require` is working properly, and proper performance test it. I would actually suggest to seperate them into two PRs. So we can merge this PR first and then carefully work on the globalpatches one.

Reply:
Fair enough. So let's limit its use in tests for the moment.